langchain/prompts/chat.py

```diff
@@ -85,6 +87,16 @@ def format_messages(self, **kwargs: Any) -> List[BaseMessage]:
     def input_variables(self) -> List[str]:
         return self.prompt.input_variables
 
+    @property
+    @abstractmethod
+    def _type(self) -> str:
+        """The type of MessagePromptTemplate."""
+
+    def dict(self, *args, **kwargs):
+        result = super().dict(*args, **kwargs)
+        result["_type"] = self._type
```

Comment:
This would change the serialization fwiw since it uses `type` with no underscore